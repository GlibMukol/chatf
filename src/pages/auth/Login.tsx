import { Input } from "@/componets/Inputs/Input"

export const Login = () => {
    return (
        <form className="relative flex flex-col justify-around items-center" autoComplete="off">
            <div>
                <Input
                    type="email"
                    id="email"
                    className='invalid:border-red-500! peer'
                    placeholder="Enter your Email"
                    required
                    pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$"
                    autoComplete="off"
                    label='Email'
                    message='Invalid Email'
                />
            </div>

            <div>
                <Input
                    type="password"
                    placeholder="Enter your password"
                    minLength={8}
                    label='Password'
                    message='Short password'
                />
            </div>
            <button type="submit">LogIn</button>
        </form>
    )
}