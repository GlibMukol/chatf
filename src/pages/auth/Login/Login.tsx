import { Input } from "@/componets/Inputs/Input"
import { useLogin } from '@/pages/auth/hooks/useLogin'

export const Login = () => {
    const {
        formRef,
        emailRef,
        passwordRef,
        disabledBtn,
        handleFormOnChange,
        submit
    } = useLogin()
    return (
        <form
            onChange={handleFormOnChange}
            onSubmit={submit}
            ref={formRef}
            className="relative flex flex-col justify-around items-center peer"
            autoComplete="off"
            noValidate
            data-testid="login form"
        >
            <div>
                <Input
                    ref={emailRef}
                    type="email"
                    id="email"
                    name="email"
                    className='invalid:border-red-500! peer'
                    placeholder="Enter your Email"
                    required
                    pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$"
                    autoComplete="off"
                    label='Email'
                    message='Invalid Email'
                    data-testid="email"
                />
            </div>
            <div>
                <Input
                    ref={passwordRef}
                    type="password"
                    name="password"
                    id="password"
                    placeholder="Enter your password"
                    minLength={8}
                    pattern=".{8,}"
                    label='Password'
                    message='Short password'
                    data-testid="password"
                    required
                />
            </div>
            <button data-testid="submit" type="submit" disabled={disabledBtn}>LogIn</button>
        </form>
    )
}