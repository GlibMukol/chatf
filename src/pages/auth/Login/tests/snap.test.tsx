import { render } from "@testing-library/react";
import { Login } from "@/pages/auth/Login/Login";
import { describe, it, expect } from "vitest"


describe("Login snapshots", () => {
    it("should correct render", () => {
        const { container } = render(<Login />);
        expect(container.firstChild).toMatchSnapshot();
    })
})
