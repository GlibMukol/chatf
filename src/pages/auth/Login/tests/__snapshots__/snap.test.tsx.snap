// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login snapshots > should correct render 1`] = `
<form
  autocomplete="off"
  class="relative flex flex-col justify-around items-center peer"
  data-testid="login form"
  novalidate=""
>
  <div>
    <label
      for="email"
    >
      Email
    </label>
    <input
      autocomplete="off"
      class="invalid:border-red-500! peer"
      data-testid="email"
      id="email"
      label="Email"
      message="Invalid Email"
      name="email"
      pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,}$"
      placeholder="Enter your Email"
      required=""
      type="email"
    />
    <p
      class="invisible text-red-500 text-sm peer-focus:invisible peer-invalid:visible transition delay-300 duration-3"
    >
      Invalid Email
    </p>
  </div>
  <div>
    <label
      for="password"
    >
      Password
    </label>
    <input
      class="invalid:border-red-500! peer"
      data-testid="password"
      id="password"
      label="Password"
      message="Short password"
      minlength="8"
      name="password"
      pattern=".{8,}"
      placeholder="Enter your password"
      required=""
      type="password"
    />
    <p
      class="invisible text-red-500 text-sm peer-focus:invisible peer-invalid:visible transition delay-300 duration-3"
    >
      Short password
    </p>
  </div>
  <button
    data-testid="submit"
    disabled=""
    type="submit"
  >
    LogIn
  </button>
</form>
`;
